### PR TITLE
Tone limit 20KHz 

### PR DESCRIPTION
**Pull request by Tech-TX**
Add 2 comments since I found out where the 5KHz upper limit on Tone output frequency was, apparently added for compatibility with the Arduino library some time in the past.

If you want the full 20KHz audio frequency range, change the two lines in Tone.cpp at line 38 to:
```cpp
  high = std::max(high, (uint32_t)25);  // new 20KHz maximum Tone frequency
  low = std::max(low, (uint32_t)25);
```

### CODE REVIEW

File: cores/esp8266/Tone.cpp [hunk 1]
```diff
@@ -35,8 +35,8 @@ static void _startTone(uint8_t _pin, uint32_t high, uint32_t low, unsigned long
 
   pinMode(_pin, OUTPUT);
 
-  high = std::max(high, (uint32_t)100);
-  low = std::max(low, (uint32_t)100);
+  high = std::max(high, (uint32_t)100);  // 5KHz maximum tone for Arduino compatibility
```
Comment (earlephilhower) [inline cores/esp8266/Tone.cpp:38]: Why not just set it to 25 here (i.e. 20Khz max)?  I think you're fine allowing a larger range than Arduino old-style core.

**Comment by Tech-TX** [comment]
Done, and done!  I'll good with removing artificial restraints on operation. :smile:
Travis CI running now.

File: cores/esp8266/Tone.cpp (hunk 1 above)
Comment (Tech-TX) [reply cores/esp8266/Tone.cpp]: I'm good with it as well.  It's in.

**Comment by Tech-TX** [comment]
FYI, I've already tested this change yesterday much farther than 20KHz.  When I tried to take it to 65535 it blew up, but I sort of expected that based on your notes elsewhere about CPU utilization at 40KHz PWM.

**Pull request merged**